.github/workflows/ci.yml

```diff
@@ -24,20 +24,114 @@ on:
 
 jobs:
   check-semantic-version:
-    if: github.event.pull_request.draft == false
+    if: github.ref != 'refs/heads/main'
     uses: octue/workflows/.github/workflows/check-semantic-version.yml@main
     with:
       path: pyproject.toml
       breaking_change_indicated_by: minor
 
-  run-tests:
+  check-main-merge:
+    if: github.ref != 'refs/heads/main'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Check branch is ahead of main
+        run: |
+          if ! git merge-base --is-ancestor origin/main ${{ github.event.pull_request.head.sha }};
+          then echo "::error::This branch is not up-to-date with the latest main branch commit.";
+          exit 1; fi
+
+  lint:
```

Comment:
Why do we need to run pre-commit both locally and remotely? This seems like an unnecessary step as the developer still has to make the changes locally, just with a longer feedback loop

Reply:
It's run remotely to ensure that any changes made in the GitHub editor pass. It also avoids the problem of having to force all contributors to use precommit, whilst enforcing the standards that are embodied.

Reply:
In what situations is the github editor used to make changes?